test/test_circ_deque.jl

```diff
@@ -80,6 +92,23 @@
         for i in 1:5 push!(D, i) end
         @test collect([i for i in D]) == collect(1:5)
     end
+
+    @testset "constructing from iterator" begin
+        # when eltype is known
+        D = CircularDeque(1:3)
+        @test_skip eltype(D) == Int
+        @test capacity(D) == 3
+        @test length(D) == 3
+
+        # when eltype is not known
+        D = CircularDeque(KnownLengthIter(10))
```

Comment:
The comment above this line isn't quite accurate.  The eltype is `Any`, which is "known".